common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkDataSource.kt

```diff
@@ -28,37 +24,59 @@ open class GetNetworkDataSource<T>(
   override suspend fun get(query: Query): T {
     return network {
       val response = when (query) {
+
+        is GenericOAuthNetworkQuery -> {
+          httpClient.get<String> {
+            oauthPasswordHeader(getPasswordTokenInteractor = query.getPasswordTokenInteractor)
+            createHttpRequestFromGenericNetworkQuery(query)
+          }
+        }
+
+        is GenericNetworkQuery -> {
+          httpClient.get<String>() {
+            createHttpRequestFromGenericNetworkQuery(query)
+          }
+        }
+
+        else -> {
+          httpClient.get<String>(_url) {
+            headers(globalHeaders)
+          }
+        }
```

Comment:
I think you could resolve all the properties required for the HTTP with the **when** expression and then you just have a single call to the **httpClient**.

If you resolve the headers, params, and the URL in the when you just need to call the httpClient outside once.

Reply:
Answered in the next one. We can close this comment